src/ClientRuntime/Microsoft.Rest.ClientRuntime.E2E.Tests/ScenarioTests/VMTests.cs

```
@@ -0,0 +1,88 @@
+            ResourceGroup resGroup = null;
+        
+            using (base.MockContext = MockContext.Start(this.GetType().FullName))
+            {
+                try
+                {
+                    string newVmId = "5C6F1669-C183-4BFC-9BBB-138E0892E917";
+                    string asName = "as5913";
+                    resGroup = CreateResourceGroup(resourceGroupName);
+                    StorageAccount storageAccount = CreateStorageAccount(resGroup, storageName);
+                    ImageReference imageRef = GetPlatformVMImage(useWindowsImage: true);
+                    VirtualMachine vm = CreateVM_NoAsyncTracking(resGroup.Name, vmName, asName, storageAccount, imageRef, out vm1);
+                    
+                    //Create a new VM and Update VmId protected Property
+                    MyVm myNewVm = new MyVm(vm);
+                    myNewVm.UpdateVm(newVmId);
+                    VirtualMachine updatedVm = myNewVm as VirtualMachine;
+
+                    // Update VM
+                    AzureOperationResponse<VirtualMachine> res = Task<AzureOperationResponse<VirtualMachine>>.Run(async () =>
+                    {
+                        return await ComputeClient.VirtualMachines.BeginCreateOrUpdateWithHttpMessagesAsync(resGroup.Name, updatedVm.Name, updatedVm).ConfigureAwait(false);
+                    }).GetAwaiter().GetResult();
+
+                    //Get Request Content and verify it does not contain VmId property
+                    string requestContentStr = ComputeClient.GetRequestContent();
+                    Assert.False(requestContentStr.Contains("VmId"));
+
+                    //Get VM Object
+                    VirtualMachine getVm = res.Body;
+
+                    // Verify the vmPutResponse does not contain updated VmId
+                    Assert.NotEqual(newVmId, getVm.VmId);
+                }
+                catch(Exception ex)
+                {
+                    Debug.WriteLine(ex.ToString());
```

@markcowl yes I should not be swallowing exception.